Booking.java

```diff
@@ -56,10 +55,33 @@ public Review getReview() {
 
     @Override
     public boolean checkConstraints() {
-        return true;
+        if(constraint_13()) return true;
+        return false;
     }
 
     public static boolean checkAllIntancesConstraints(Model model){
+        for(Object o: model.allObjects){
+            if(o instanceof Booking){
+                if(!((Booking) o).checkConstraints()) return false;
+            }
+        }
         return true;
     }
+
+    /**
+     * All services given in the booking belong to the hosting hotel.
+     * @return
+     */
+    public boolean constraint_13(){
+        HashMap<Service, HotelService> hotelServices = this.room.getHotel().getServices();
+        for(HotelService service: this.services) {
+            boolean found = false;
+            for (HotelService hotelService : hotelServices.values()) {
+                if (service == hotelService) found = true;
+            }
+            if (!found) return false;
+        }
+        return true;
+
+    }
```

Comment:
you sure that this is constraint 13?
you need to compare the hotels, no?

Reply:
It's constraint number 13 in part B.
Anyways, the constraint checks the services in a specific booking(which is booked to a specific hotel) are given by this specific hotel. In line 76 it get's a specific hotel's services - which means there is a comparison thus there shouldn't be another comparison.